architecture-decision-records/0037-disable-inactive-keycloak-users.md

```diff
@@ -0,0 +1,37 @@
+# 37. Disable inactive keycloak users
+
+**Date**: 2025-02-17
+
+## Context
+We need a reliable mechanism to identify and disable users who have not logged in for x about of days. Two main approaches were considered:
+1. Using Keycloak events to track user activity
+2. Directly querying application database for user activity and cross-referencing with Keycloak
+
+The system needs to ensure that inactive users are properly disabled for security and resource management purposes.
+
+## Decision
+We will implement a user deactivation strategy based on querying application database for user activity and cross-referencing with Keycloak rather than relying on solely on Keycloak events. The approach will:
+
+1. Query the application database to identify users who haven't been active for a specified period (x days)
```

Comment:
Rephrase slightly: `... users who have been active for ...`

Need to get the active users from the TDR application database. Consider the scenario of a user who has never done a transfer. There would be no record of such a user in the TDR application database. When you compare the list with Keycloak, that user would not be disabled.